packets.go

```diff
@@ -525,6 +532,13 @@ func (mc *mysqlConn) handleErrorPacket(data []byte) error {
 		pos = 9
 	}
 
+	//If error code is for Connection was killed, then return bad connection.
+	//https://mariadb.com/kb/en/mariadb/mariadb-error-codes/
+	if errno == 1927 {
+		errLog.Print("Error ", errno, ": ", string(data[pos:]))
+		return driver.ErrBadConn
```

Comment:
You can close this connection (`mc.close()` to make next call returns `ErrBadConn`.